### DISCUSSION TITLE
Per workspace extensions

### DISCUSSION THREAD

**Discussion by fusion44** [zen-browser/desktop#2219] [General]
Hi!

I'm looking into whether I can switch to Zen from Firefox. The first experience in Zen is wonderful.

One problem I have is that I do use WakaTime to track my time I spend working on my various projects. Unfortunately, the WakaTime extension allows to set only one project setting per browser profile. I really would like to avoid having to switch between profiles to fix this. Note, this is not a Zen specific problem.

I'm looking into using workspaces to improve this. As far as I can see, there is no way yet to override settings of extensions for each workspace. Is creating a Zen specific extension for WakaTime be a viable solution? Is the workspaces API documented somewhere?

**Comment by Fxxxlei** [comment]
I'm not sure Zen Browser actually currently provides any additional API for extensions. Zen plugins are planned but I wouldn't count on it anytime soon, since that sounds like a lot of work.

Would it be viable to use containers (potentially linked to workspaces) for your use case? I know they do a lot of other stuff you might not want, but since they are a Firefox feature, there is an API to interact with them.

**Reply by fusion44** [reply]
As far as I understand it is a Firefox browser limitation that you can't easily switch profiles without restarting the browser, or starting a new process. I'm currently rethinking of how I can actually do that from withing my system configuration.

From my limited experience with developing extensions for browsers it doesn't seem too much work to integrate an existing JS code base into ZEN. I just doubt how much sense it makes to integrate it into the browser itself. The extension is not used by many people and yet it's delivered to everybody.

**Comment by fusion44** [comment]
I'll close this as there are other, more active, discussions that are about the same topic.
Example: https://github.com/zen-browser/desktop/discussions/2337

**Comment by SamuXzX** [comment]
This would make the workspace feature much more powerful, but with the recent displacement of extensions from extensions from the extensions menu to the "unified panel" in the address bar it could seem incoherent: currently extensions seems related to the tab because they are placed in the address bar; if we had the old extensions menu they could much better be thought as workspace-specific. I talk about this here: https://github.com/zen-browser/desktop/discussions/11014
